axelrod/ecosystem.py

```diff
@@ -0,0 +1,59 @@
+
+import random
+
+
+class Ecosystem(object):
+    """Create an ecosystem based on the payoff matrix from an Axelrod tournament."""
+
+    def __init__(self, results, fitness=None, ):
+
+        results.init_output()
+        self.results = results
+        self.nplayers = self.results.nplayers
+        self.payoff_matrix = self.results.payoff_matrix
+        self.payoff_stddevs = self.results.payoff_stddevs
+
+        # Population sizes will be recorded in this nested list, with each internal
+        # list containing strategy populations for a given turn. The first list,
+        # representing the starting populations, should have all equal values,
+        # and all population lists should be normalized to one.
+        self.population_sizes = [[1.0 / self.nplayers for i in range(self.nplayers)]]
+
+        # This function is quite arbitrary and probably only influence the kinetics
+        # for the current code.
+        if fitness:
+            self.fitness = fitness
+        else:
+            self.fitness = lambda p: 5.0 / (5.0 + p)
+
+    def reproduce(self, turns):
+
+        for iturn in range(turns):
+
+            plist = list(range(self.nplayers))
+            pops = self.population_sizes[-1]
+
+            # The unit payoff for each player in this turn is the sum of the payoffs
+            # obtained from playing with all other players, scaled by the size of the
+            # opponent's population. Note that we sample the normal distribution
+            # based on the payoff matrix and its standard deviations obtained from
+            # the iterated PD tournament run previously.
+            payoffs = [0 for ip in plist]
+            for ip in plist:
+                for jp in plist:
+                    avg = self.payoff_matrix[ip][jp]
+                    dev = self.payoff_stddevs[ip][jp]
+                    p = random.normalvariate(avg, dev)
```

Comment:
This wasn't what I meant by the noise/mutation. I'm happy for you to revert this back to where it is (does no harm though so we can leave it in I suppose... Your call.

(Happy to discuss what I meant but I think we can just leave this PR as deterministic).